pkg/daemon/ceph/client/mirror_health.go

```diff
@@ -0,0 +1,217 @@
+/*
+Copyright 2020 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+import (
+	"context"
+	"time"
+
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	"github.com/rook/rook/pkg/clusterd"
+	"github.com/rook/rook/pkg/operator/ceph/reporting"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	defaultHealthCheckInterval = 1 * time.Minute
+)
+
+type mirrorChecker struct {
+	context        *clusterd.Context
+	interval       *time.Duration
+	client         client.Client
+	clusterInfo    *ClusterInfo
+	namespacedName types.NamespacedName
+	monitoringSpec *cephv1.NamedPoolSpec
+	objectType     client.Object
+}
+
+// newMirrorChecker creates a new HealthChecker object
+func NewMirrorChecker(context *clusterd.Context, client client.Client, clusterInfo *ClusterInfo, namespacedName types.NamespacedName, monitoringSpec *cephv1.NamedPoolSpec, object client.Object) *mirrorChecker {
+	c := &mirrorChecker{
+		context:        context,
+		interval:       &defaultHealthCheckInterval,
+		clusterInfo:    clusterInfo,
+		namespacedName: namespacedName,
+		client:         client,
+		monitoringSpec: monitoringSpec,
+		objectType:     object,
+	}
+
+	// allow overriding the check interval
+	checkInterval := monitoringSpec.StatusCheck.Mirror.Interval
+	if checkInterval != nil {
+		logger.Infof("mirroring status check interval for pool/radosnamespace %q is %q", namespacedName.Name, checkInterval.Duration.String())
+		c.interval = &checkInterval.Duration
+	}
+
+	return c
+}
+
+// checkMirroring periodically checks the health of the cluster
+func (c *mirrorChecker) CheckMirroring(context context.Context) {
+	// check the mirroring health immediately before starting the loop
+	err := c.CheckMirroringHealth()
+	if err != nil {
+		c.UpdateStatusMirroring(nil, nil, nil, err.Error())
+		logger.Debugf("failed to check pool/radosnamespace mirroring status for pool/radosnamespace %q. %v", c.namespacedName.Name, err)
+	}
+
+	for {
+		select {
+		case <-context.Done():
+			logger.Infof("stopping monitoring pool/radosnamespace mirroring status %q", c.namespacedName.Name)
```

Comment:
for all these error and log messages, let's not include "pool/radosnamespace" everywhere. It just makes the messages harder to read. 
```suggestion
			logger.Infof("stopping monitoring mirroring status for %q", c.namespacedName.Name)
```